pkg/scheduler/actions/integration_tests/allocate/allocate_topology_test.go

```diff
@@ -0,0 +1,1055 @@
+							Priority:            constants.PriorityTrainNumber,
+							QueueName:           "queue0",
+							Tasks: []*tasks_fake.TestTaskBasic{
+								{
+									State:    pod_status.Running,
+									NodeName: "node0",
+								},
+							},
+						},
+						{
+							Name:                "running_job1",
+							RequiredGPUsPerTask: 1,
+							Priority:            constants.PriorityTrainNumber,
+							QueueName:           "queue0",
+							Tasks: []*tasks_fake.TestTaskBasic{
+								{
+									State:    pod_status.Running,
+									NodeName: "node1",
+								},
+							},
+						},
+						{
+							Name:                "pending_job0",
+							RequiredGPUsPerTask: 1,
+							Priority:            constants.PriorityTrainNumber,
+							QueueName:           "queue0",
+							Topology: &enginev2alpha2.TopologyConstraint{
+								Topology:               "cluster-topology",
+								PreferredTopologyLevel: "k8s.io/rack",
+							},
+							Tasks: []*tasks_fake.TestTaskBasic{
+								{
+									State: pod_status.Pending,
+								},
+								{
+									State: pod_status.Pending,
+								},
+							},
+						},
+					},
+					Nodes: map[string]nodes_fake.TestNodeBasic{
+						"node0": {
+							GPUs: 2,
+							Labels: map[string]string{
+								"k8s.io/rack": "rack1",
+							},
+						},
+						"node1": {
+							GPUs: 2,
+							Labels: map[string]string{
+								"k8s.io/rack": "rack2",
+							},
+						},
+					},
+					Queues: []test_utils.TestQueueBasic{
+						{
+							Name:               "queue0",
+							ParentQueue:        "department-a",
+							DeservedGPUs:       4,
+							GPUOverQuotaWeight: 1,
+							MaxAllowedGPUs:     4,
+						},
+					},
+					Departments: []test_utils.TestDepartmentBasic{
+						{
+							Name:         "department-a",
+							DeservedGPUs: 4,
+						},
+					},
+					TaskExpectedResults: map[string]test_utils.TestExpectedResultBasic{
+						"running_job0-0": {
+							GPUsRequired:         1,
+							Status:               pod_status.Running,
+							DontValidateGPUGroup: true,
+						},
+						"running_job1-0": {
+							GPUsRequired:         1,
+							Status:               pod_status.Running,
+							DontValidateGPUGroup: true,
+						},
+						"pending_job0-0": {
+							GPUsRequired:         1,
+							Status:               pod_status.Running,
+							DontValidateGPUGroup: true,
+						},
+						"pending_job0-1": {
+							GPUsRequired:         1,
+							Status:               pod_status.Running,
+							DontValidateGPUGroup: true,
+						},
+					},
+					Mocks: &test_utils.TestMock{
+						CacheRequirements: &test_utils.CacheMocking{
+							NumberOfCacheBinds: 2,
+						},
+					},
+				},
+				RoundsUntilMatch: 1,
+			},
+		*/
+		{
+			TestTopologyBasic: test_utils.TestTopologyBasic{
+				Name: "Prefer + Require topology level - schedule all tasks on capable require domain (even though first require domain capable to have part of the tasks), spread on multiple prefer domain",
```

Comment:
Suggesting a shorter name:
"Combined preferred and required topology constraints - schedule job although preferred constraint cannot be satisfied"